### ISSUE TITLE
Restrict completion items based on prefix

### ISSUE THREAD

**Issue by jfacorro** [erlang-ls/erlang_ls#203]
**Is your feature request related to a problem? Please describe.**
When trying out the branch that indexes the application, the deps and OTP, the completion feature made typing anything extremely slow. This is in part because the completion currently returns all indexed modules, and when there are *a lot* this means the full list is returned on every completion.

**Describe the solution you'd like**
Use the prefix of what has been typed so far to filter the items returned in the completion.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by robertoaloi** [comment]
I noticed this too and thought about fixing it as part of the branch, but then decided it was out of scope.
Please hold on on this for now to avoid conflicts with the branch.

**Comment by robertoaloi** [comment]
Depends on #183

**Comment by jfacorro** [comment]
☝️ That is implemented by branching off the indexing branch.

**Comment by jfacorro** [comment]
> Depends on #183
I don't think it really depends on that issue, since they can be implemented separately,  right?

**Comment by robertoaloi** [comment]
I am simply referring to the fact there will be conflicts if we introduce this prior to the indexing branch (e.g. the index name has been renamed). Not a big a deal.
